docs/tutorial/part_1.rst

```diff
@@ -0,0 +1,238 @@
+
+        def __str__(self):
+            return f"Room({self.name} {self.host})"
+        
+    class Message(models.Model):
+        room = models.ForeignKey("chat.Room", on_delete=models.CASCADE, related_name="messages")
+        text = models.TextField(max_length=500)
+        user = models.ForeignKey(User, on_delete=models.CASCADE, related_name="messages")
+
+        def __str__(self):
+            return f"Message({self.user} {self.room})"
+        
+Creating the Serializers
+------------------------
+
+We will put the following code in the ``serializers.py`` file, to handle the serialization of the models created.
+
+.. code-block:: python
+
+    from .models import User, Room, Message
+    from rest_framework import serializers
+
+    class UserSerializer(serializers.ModelSerializer):
+        class Meta:
+            model = User
+            exclude = ["password"]
+
+    class MessageSerializer(serializers.ModelSerializer):
+        created_at_formatted = serializers.SerializerMethodField()
+        user = UserSerializer()
+        class Meta:
+            model = Message
+            exclude = []
+            depth = 1
+
+        def get_created_at_formatted(self, obj:Message):
+            return obj.created_at.strftime("%d-%m-%Y %H:%M:%S")
+
+    class RoomSerializer(serializers.ModelSerializer):
+        last_message = serializers.SerializerMethodField()
+        messages = MessageSerializer(many=True, read_only=True)
+        class Meta:
+            model = Room
+            fields = ["pk", "name", "host", "messages", "current_users", "last_message"]
+            depth = 1
+            read_only_fields = ["messages", "last_message"]
+            
+        def get_last_message(self, obj:Room):
+            return MessageSerializer(obj.messages.order_by('created_at').last()).data
+
+
+Creating the Consumers
+----------------------
+
+In the ``consumers.py`` file we will create only the room consumer for:
+    * Joining and leaving a room.
+    * Observing messages in that room.
+    * Observing the current users in the room.
+
+.. code-block:: python
+
+    import json
+    from django.shortcuts import get_object_or_404
+    from channels.generic.websocket import AsyncWebsocketConsumer
+    from channels.db import database_sync_to_async
+    from django.utils.timezone import now
+    from django.conf import settings
+    from typing import Generator
+    from djangochannelsrestframework.generics import GenericAsyncAPIConsumer, AsyncAPIConsumer
+    from djangochannelsrestframework.observer.generics import (ObserverModelInstanceMixin, action)
+    from djangochannelsrestframework.observer import model_observer
+
+    from .models import Room, Message, User
+    from .serializers import MessageSerializer, RoomSerializer, UserSerializer
+
+    class RoomConsumer(ObserverModelInstanceMixin, GenericAsyncAPIConsumer):
+        queryset = Room.objects.all()
+        serializer_class = RoomSerializer
+        lookup_field = "pk"
```

Comment:
might be nice to add `room_subscribe: Optional[Int] = None` or something so that we are not add hock adding attributes to the class a runtime.